### PR TITLE
Added support for insecure OCI registries

### PR DESCRIPTION
**Pull request by sabre1041**
Signed-off-by: Andrew Block <andy.block@gmail.com>

**What this PR does / why we need it**:

Supports connectivity for insecure OCI registries. 

Relates to #11352

### CODE REVIEW

File: pkg/getter/httpgetter.go [hunk 1]
```diff
@@ -123,8 +123,8 @@ func (g *HTTPGetter) httpClient() (*http.Client, error) {
 		}
 	})
 
-	if (g.opts.certFile != "" && g.opts.keyFile != "") || g.opts.caFile != "" {
-		tlsConf, err := tlsutil.NewClientTLS(g.opts.certFile, g.opts.keyFile, g.opts.caFile)
+	if (g.opts.certFile != "" && g.opts.keyFile != "") || g.opts.caFile != "" || g.opts.insecureSkipVerifyTLS {
```
Comment (souleb) [inline pkg/getter/httpgetter.go:126]: this should be another PR imo.

File: pkg/pusher/ocipusher_test.go [hunk 2]
```diff
@@ -35,10 +35,12 @@ func TestNewOCIPusher(t *testing.T) {
 	cd := "../../testdata"
 	join := filepath.Join
 	ca, pub, priv := join(cd, "rootca.crt"), join(cd, "crt.pem"), join(cd, "key.pem")
+	insecureSkipTLSverify := false
 
 	// Test with options
 	p, err = NewOCIPusher(
 		WithTLSClientConfig(pub, priv, ca),
+		WithInsecureSkipTLSVerify(insecureSkipTLSverify),
```
Comment (souleb) [inline pkg/pusher/ocipusher_test.go:43]: can you add this in `ocigetter_test.go` as well?

**Comment by sabre1041** [comment]
@souleb for the comment related to the `registry_login.go` file, did you mean for it to reference the `action` package instead of `cmd`?

**Comment by souleb** [comment]
> ```go
> 		registry.LoginOptInsecure(insecure),
> ```

I mean that we should use the `insecureSkipTLSVerify` flag there too, and use the option `registry.LoginOptInsecure(insecure)` when login.

**Comment by sabre1041** [comment]
@souleb got it. There is no actual functionality in the login process aside from the flag option, so I will work on integrating that in as part of this PR

**Comment by souleb** [comment]
`pkg/getter/httpgetter.go` should be tested I think and have its own PR.

**Comment by sabre1041** [comment]
@souleb is it just the enhancements to `pkg/getter/httpgetter.go` that should be in a separate PR? Just trying to determine where to split it up

**Comment by souleb** [comment]
> @souleb is it just the enhancements to `pkg/getter/httpgetter.go` that should be in a separate PR? Just trying to determine where to split it up

Yes. To me that addresses `insecure` for `https` repositories.

**Comment by souleb** [comment]
To me here we adresses OCI registries hosting helm artifacts. An `httpgetter` is used to access a helm repository (with a index file). So I would say it's out of scope.

**Comment by sabre1041** [comment]
@souleb made the requested changes

**Review by souleb — APPROVED** [review]
LGTM!

**Comment by keskad** [comment]
Tested it, seems to be working.

https://github.com/helm/helm/issues/11683#issuecomment-1378482533
